node/pkg/query/query.go

```diff
@@ -0,0 +1,412 @@
+		chainQueryReqC       map[vaa.ChainID]chan *PerChainQueryInternal
+		queryResponseReadC   <-chan *PerChainQueryResponseInternal
+		queryResponseWriteC  chan<- *QueryResponsePublication
+		allowedRequestors    map[ethCommon.Address]struct{}
+	}
+
+	// pendingQuery is the cache entry for a given query.
+	pendingQuery struct {
+		signedRequest *gossipv1.SignedQueryRequest
+		request       *QueryRequest
+		requestID     string
+		receiveTime   time.Time
+		queries       []*perChainQuery
+		responses     []*PerChainQueryResponseInternal
+
+		// respPub is only populated when we need to retry sending the response to p2p.
+		respPub *QueryResponsePublication
+	}
+
+	// perChainQuery is the data associated with a single per chain query in a query request.
+	perChainQuery struct {
+		req            *PerChainQueryInternal
+		channel        chan *PerChainQueryInternal
+		lastUpdateTime time.Time
+	}
+)
+
+// Start initializes the query handler and starts the runnable.
+func (qh *QueryHandler) Start(ctx context.Context) error {
+	qh.logger.Debug("entering Start", zap.String("enforceFlag", qh.allowedRequestorsStr))
+
+	var err error
+	qh.allowedRequestors, err = parseAllowedRequesters(qh.allowedRequestorsStr)
+	if err != nil {
+		return fmt.Errorf("failed to parse allowed requesters: %w", err)
+	}
+
+	if err := supervisor.Run(ctx, "query_handler", common.WrapWithScissors(qh.handleQueryRequests, "query_handler")); err != nil {
+		return fmt.Errorf("failed to start query handler routine: %w", err)
+	}
+
+	return nil
+}
+
+// handleQueryRequests multiplexes observation requests to the appropriate chain
+func (qh *QueryHandler) handleQueryRequests(ctx context.Context) error {
+	return handleQueryRequestsImpl(ctx, qh.logger, qh.signedQueryReqC, qh.chainQueryReqC, qh.allowedRequestors, qh.queryResponseReadC, qh.queryResponseWriteC, qh.env, RequestTimeout, RetryInterval)
+}
+
+// handleQueryRequestsImpl allows instantiating the handler in the test environment with shorter timeout and retry parameters.
+func handleQueryRequestsImpl(
+	ctx context.Context,
+	logger *zap.Logger,
+	signedQueryReqC <-chan *gossipv1.SignedQueryRequest,
+	chainQueryReqC map[vaa.ChainID]chan *PerChainQueryInternal,
+	allowedRequestors map[ethCommon.Address]struct{},
+	queryResponseReadC <-chan *PerChainQueryResponseInternal,
+	queryResponseWriteC chan<- *QueryResponsePublication,
+	env common.Environment,
+	requestTimeoutImpl time.Duration,
+	retryIntervalImpl time.Duration,
+) error {
+	qLogger := logger.With(zap.String("component", "ccqhandler"))
+	qLogger.Info("cross chain queries are enabled", zap.Any("allowedRequestors", allowedRequestors), zap.String("env", string(env)))
+
+	pendingQueries := make(map[string]*pendingQuery) // Key is requestID.
+
+	// TODO: This should only include watchers that are actually running. Also need to test all these chains.
+	supportedChains := map[vaa.ChainID]struct{}{
+		vaa.ChainIDEthereum:  {},
```

Comment:
Do you need to call the shouldStart()-like functions here?  Or this feature isn't there, yet?

Reply:
I changed the code to use one of the maps passed in to remove chains that are not enabled.